kubernetes/e2e_test/test_utils.py

```diff
@@ -17,100 +17,282 @@
-        dep = dep_api.read_namespaced_deployment(name="nginx-deployment", 
-            namespace="dep")
+        dep_api = utils.create_from_yaml(
+            k8s_client, "kubernetes/e2e_test/test_yaml/extensions-deployment-dep.yaml")
+        dep = dep_api.read_namespaced_deployment(name="nginx-deployment",
+                                                 namespace="dep")
         self.assertIsNotNone(dep)
-        resp = dep_api.delete_namespaced_deployment(
-            name="nginx-deployment", namespace="dep", 
+        dep_api.delete_namespaced_deployment(
+            name="nginx-deployment", namespace="dep",
             body={})
-        resp = core_api.delete_namespace(name="dep", body={})
-        
-    def test_api_service(self):
+        core_api.delete_namespace(name="dep", body={})
+
+    def test_api_service_with_conflict(self):
+        """
+        Test: Create an APIService. Recreate the APIService to test conflict
+        handling. Delete the APIService.
+        Check the program raises CreationFailedError.
+        Using apiregistration API
+        Using yaml file api-service.yaml
+        """
         k8s_client = client.api_client.ApiClient(configuration=self.config)
-        k8s_api = utils.create_from_yaml(k8s_client, 
-            "kubernetes/e2e_test/test_yaml/api-service.yaml")
-        self.assertEqual("apiregistration.k8s.io/v1beta1", 
-            k8s_api.get_api_resources().group_version)
+        k8s_api = utils.create_from_yaml(
+            k8s_client, "kubernetes/e2e_test/test_yaml/api-service.yaml")
+        self.assertEqual("apiregistration.k8s.io/v1beta1",
+                         k8s_api.get_api_resources().group_version)
         svc = k8s_api.read_api_service(
             name="v1alpha1.wardle.k8s.io")
         self.assertIsNotNone(svc)
-        resp = k8s_api.delete_api_service(
-            name="v1alpha1.wardle.k8s.io", body={})
\ No newline at end of file
+        with self.assertRaises(utils.FailToCreateError):
+            utils.create_from_yaml(
+                k8s_client, "kubernetes/e2e_test/test_yaml/api-service.yaml")
+        k8s_api.delete_api_service(
+            name="v1alpha1.wardle.k8s.io", body={})
+
+    def test_list(self):
+        """
+        Test for creating and deleting a service and a deployment using
+        the List kind.
+        Using core, extensions API
+        Using yaml file list.yaml
+        """
+        k8s_client = client.api_client.ApiClient(configuration=self.config)
+        k8s_api = utils.create_from_yaml(
+            k8s_client, "kubernetes/e2e_test/test_yaml/list.yaml")
+        svc_api = k8s_api[0]
+        self.assertEqual("v1", svc_api.get_api_resources().group_version)
+        svc = svc_api.read_namespaced_service(name="list-service-test",
+                                              namespace="default")
+        self.assertIsNotNone(svc)
+        ext_api = k8s_api[1]
```

Comment:
I don't see the point of doing so. It's typed client anyway

it's not a good pattern to iterate through a list of arbitrary api objects and call typed api functions imo